src/extensions/default/InAppNotifications/main.js

```diff
@@ -0,0 +1,372 @@
+        }
+
+        // PRERELEASE_BEGIN
+        // The following code is needed for supporting notifications in prerelease,
+        {
+            if (locale) {
+                if(locale.length > 2) {
+                    locale = locale.substring(0, 2);
+                }
+                switch(locale)  {
+                case "de":
+                    break;
+                case "es":
+                    break;
+                case "fr":
+                    break;
+                case "ja":
+                    break;
+                case "en":
+                default:
+                    locale = "en";
+                }
+                return brackets.config.notification_info_url.replace("<locale>", locale);
+            }
+        }
+        // PRERELEASE_END
+
+        return brackets.config.notification_info_url + '?locale=' + locale;
+    }
+
+    /**
+     * Get a data structure that has information for all Brackets targeted notifications.
+     *
+     * _versionInfoUrl is used for unit testing.
+     */
+    function _getNotificationInformation(_versionInfoUrl) {
+        // Last time the versionInfoURL was fetched
+        var lastInfoURLFetchTime = PreferencesManager.getViewState("lastNotificationURLFetchTime");
+
+        var result = new $.Deferred();
+        var fetchData = false;
+        var data;
+
+        // If we don't have data saved in prefs, fetch
+        data = PreferencesManager.getViewState("notificationInfo");
+        if (!data) {
+            fetchData = true;
+        }
+
+        // If more than 24 hours have passed since our last fetch, fetch again
+        if ((new Date()).getTime() > lastInfoURLFetchTime + ONE_DAY) {
+            fetchData = true;
+        }
+
+        if (fetchData) {
+            var lookupPromise = new $.Deferred(),
+                localVersionInfoUrl;
+
+            // If the current locale isn't "en" or "en-US", check whether we actually have a
+            //   locale-specific notification target, and fall back to "en" if not.
+            // Note: we check for both "en" and "en-US" to watch for the general case or
+            //    country-specific English locale.  The former appears default on Mac, while
+            //    the latter appears default on Windows.
+            var locale = brackets.getLocale().toLowerCase();
+            if (locale !== "en" && locale !== "en-us") {
+                localVersionInfoUrl = _versionInfoUrl || _getVersionInfoUrl();
+                $.ajax({
+                    url: localVersionInfoUrl,
+                    cache: false,
+                    type: "HEAD"
+                }).fail(function (jqXHR, status, error) {
+                    // get rid of any country information from locale and try again
+                    var tmpUrl = _getVersionInfoUrl(brackets.getLocale(), true);
+                    if (tmpUrl !== localVersionInfoUrl) {
+                        $.ajax({
+                            url: tmpUrl,
```

Comment:
Is there some way to reduce the nesting of promises?

Reply:
Didn't have time to refactor the code, copied from Update notification module.

Reply:
Done. no more "en-US" check first.